### ISSUE TITLE
Confusion between root password and SSH password

### ISSUE THREAD

**Issue by mykaul** [lago-project/lago#381]
The code is quite mixed up between the two. My understanding is:
- SSH password is needed by Lago for VMs which it did not inject the SSH public key into (for example, hosted engine VM in ovirt-system-tests testing Hosted Engine). **Which means that the default SSH password can go away.**
- Root password is not needed for Lago, but for tests on top of Lago (for example, adding host to oVirt in ovirt-system-tests). 
- Having a root password /probably/ implies having the SSH password - but it's not strictly needed (due to above injection) by Lago.
- Root password does not need to be set, as clearly all lago-images have the root password both set and known in their metadata (but Lago does not bother to read the root password from the metadata). **Which means that the default root password can go away.**

I'd like to ensure that:
1. We can use public-key anywhere it's feasible - without root password (Engine VM in hosted-engine, for example)
2. We can add ssh-password to whoever VM need it and use it (Hosted Engine VM for example).
3. We can add on top of the public key the root password, but let the tests use it and not Lago.

Alternatively, since we do have the root password always, we can drop all the code that is doing the public-key injection, as clearly we've worked nicely without it. Specifically, let's not bother creating the public key and injecting it.

@nvgoldin , @david-caro , @ifireball , @eedri 
Thoughts?

**Comment by nvgoldin** [comment]
> Root password does not need to be set, as clearly all lago-images have the root password both set and known in their metadata (but Lago does not bother to read the root password from the metadata). Which means that the default root password can go away.

I don't think we should assume that it will always be known - especially if we take images directly from virt-builder in the future. It is quite handy that VMs can be connected out-of-the-box with plain passwords as a fall-back(needed to do that  few times when connecting from a different user or with SSH tunnelling).

>Root password is not needed for Lago, but for tests on top of Lago (for example, adding host to oVirt in ovirt-system-tests).

maybe add host can be done with SSH key instead(I remember vaguely an option in the UI to add an SSH key instead, need to check if it can be done in the SDK and  how we can retrieve it during the tests).

I think we should:
1. Inject SSH keys and ensure a password connection is only a fallback(which might be a little hard with paramiko's various exceptions), if it uses the root password to connect, issue a warning.
2. Always change the root password, unless the user specificity asks not(maybe as 'noset' value in root_password  config parameter, or explicitly in the yaml file). 
3. Attempt to use SSH keys also in the tests. This means improving how we inject SSH keys, we could:  
   1. Generate ssh-pair for each VM.
   2. Inject to each VM its own key and all other VMs public key to authorized_keys file.

This will need some thought on how to do it in parallel.
In the future - adding 'cloud-init' support here would be useful, bypassing the need to do this in virt-sysprep. Alternatively, we could relay on fixed SSH pairs in the tests and add them to them init file.


>Alternatively, since we do have the root password always, we can drop all the code that is doing the public-key injection and use, as clearly we've worked nicely without it. Specifically, let's not bother creating the public key and injecting it.

I'm mixed here - on one side, its true, security is not a true concern at the moment and it takes some effort to setup SSH keys properly. On the other hand, SSH might come handy in the future..
